lib/datadog/appsec/contrib/rails/gateway/request.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require_relative '../../../instrumentation/gateway/argument'
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Rails
+        module Gateway
+          # Gateway Request argument. Normalized extration of data from ActionDispatch::Request
+          class Request < Instrumentation::Gateway::Argument
```

Comment:
Would it make sense to inherit from `Contrib::Rack::Gateway::Request`?

Reply:
At first, I thought about, but since I saw we use the `Rack::RequestBody` directly in the [Sinatra contrib folder](https://github.com/DataDog/dd-trace-rb/pull/2648/files#diff-b20f56877521f6a704cab6694c422e0bb66bad6eff3d0d9b7b05188aa976a0dbR49) I thought it would be ok. 

I'm happy to change it.